cmd/devp2p/internal/ethtest/chain.go

```diff
@@ -68,6 +84,43 @@ func (c *Chain) Head() *types.Block {
 	return c.blocks[c.Len()-1]
 }
 
+func (c *Chain) GetHeaders(req GetBlockHeaders) (BlockHeaders, error) {
+	if req.Amount < 1 {
+		return nil, fmt.Errorf("no block headers requested")
+	}
+
+	headers := make(BlockHeaders, req.Amount)
+	var blockNumber uint64
+
+	// range over blocks to check if our chain has the requested header
+	for _, block := range c.blocks {
+		if block.Hash() == req.Origin.Hash || block.Number().Uint64() == req.Origin.Number {
+			headers[0] = block.Header()
+			blockNumber = block.Number().Uint64()
+		}
+	}
+	if headers[0] == nil {
+		return nil, fmt.Errorf("no headers found for origin") // TODO fix err
+	}
+
+	if req.Reverse {
+		for i := 1; i < int(req.Amount); i++ {
+			blockNumber -= (1 - req.Skip)
+			headers[i] = c.blocks[blockNumber].Header()
+
+		}
+
+		return headers, nil
+	}
+
+	for i := 1; i < int(req.Amount); i++ {
+		blockNumber += (1 + req.Skip)
+		headers[i] = c.blocks[blockNumber].Header()
```

Comment:
What if we don't have it? That will be a nil deref, right?

Reply:
yes out of range. But I can't really see a circumstance in an isolated test (where my test is the only peer) that a node should request block headers from me that I have not announced / the node does not know about. We have a test (`Broadcast`) which does test that a block announcement will be adequately handled by the node in which case the test suite will update its own chain with the announced block (and would technically be able to provide the header if the node asks)

Reply:
Yeah, I guess it's fine -- if we notice it panicing we can fix it later